config/v2/validation.go

```diff
@@ -91,6 +87,41 @@ func ValidateAWSProvider(p *AWSProvider, component string) error {
 	return errs
 }
 
+// ValidateBackend will check the resolved configuration for a backend and return any errors it
+// finds
+func ValidateBackend(backend *Backend, component string) error {
+	if backend == nil {
+		return nil
+	}
+
+	var errs *multierror.Error
+
+	if backend.Kind == nil {
```

Comment:
wonder if this would read nicer if each of the kind statements returned their errors rather than having elifs and the return all the way at the bottom

Reply:
yeah sgtm